gems/aws-sdk-cognitoidentity/spec/cognito_identity_credentials_spec.rb

```diff
@@ -0,0 +1,126 @@
+
+require_relative 'spec_helper'
+
+module Aws
+  module CognitoIdentity
+    describe CognitoIdentityCredentials do
+
+      let(:client) {
+        CognitoIdentity::Client.new(
+          region: 'us-west-2',
+          stub_responses: true
+        )
+      }
+
+      let(:in_one_hour) { Time.now + 60 * 60 }
+
+      let(:expiration) { in_one_hour }
+
+      let(:cognito_creds) {
+        double('credentials',
+               access_key_id: 'akid',
+               secret_key: 'secret',
+               session_token: 'session',
+               expiration: expiration
+               )
+      }
+
+      let(:identity_id) { 'identity_id' }
+      let(:identity_pool_id) { 'pool_id' }
+
+      let(:resp) { double('client-resp', credentials: cognito_creds) }
+
+      describe '#initialize' do
+
+        it 'constructs a client with passed arguments when not given' do
+          expect(CognitoIdentity::Client).to receive(:new)
```

Comment:
By the way, no need to mock clients. I think in many of our tests we actually just make them.

Reply:
Here I was testing explicitly for new being called on the client.